### ISSUE TITLE
Multiple lespas folders for easier sharing workflow / family album 

### ISSUE THREAD

**Issue by steviehs** [scubajeff/lespas#70]
(As you already know) I have organized my digital memories in YYYY/MM/(optional Event) Folders, which are flattened by my [phonepicy](https://gitlab.com/steviehs/stevemisc/-/blob/master/phonepics.py) script. 

Right now I have a flattened folder list with approx 400 entries, one new every month at least.

Some of my albums are only for my, some of them are shared with my family, which is a group on nextcloud. So have expanded the phonepics script to add sharing on Nextcloud via it's api if this is demanded. 

As you can imagine, it looks really ugly in the root folder of the family member and he has to move all this shares in a lespas folder on his side (and this also once a month for the new folder)

An alternative would be to setup the "shared_with_me" folder to lespas in Nextcloud's configuration, but this would irritate users who receives other shares than pictures. 

I tried to play around with group folders and more granular permission inside, but it seems rather complicated to script this. 

The easiest way to achieve this behaviour would be - from my perspective - to support multiple lespas folders via the app. Maybe just a regex in finding the folders would be sufficient, so lespas would not only look for the lespas folder but for any lespas* folder. With this, every family member could easily share his pictures with the rest of the family and does not need to setup complicated sharing stuff. 

Do you think there will be a chance for that? I know it has been discussed in the past, but I wanted to get around it with no success with the existing means :-)

**Comment by steviehs** [comment]
Providing a lot of albums to my family would end up for them in a mess in their root directory - or in the shared_with_me directory. To get those pictures - as I understand this - in their personal lespas apps on their devices, the members have to move the folders to a lespas folder in their root.
As a workaround, I could globally set the name of the shared_with_me directory to lespas.  With the drawback, that users have to look for other stuff they get shared in this directory... 
So, I'd say it is preferable for the normal workflow not to have a shared_with_me directory. So, it would be great if lespas would look for more than only the lespas folder but for any lespas* folder in /

In the new version, I see the icon "published from friends" on the top. How does this work?

**Comment by scubajeff** [comment]
The icon on the toolbar of the first screen is for Publication action. When the icon is grey out, it means Les Pas can't fetch friend's publication from server (most likely it's a temporary network glitch). if it's green, then your friends have updated their publications. Click on it will bring you to the list of your friend's publication, e.g., shares to you on NC server.

If all you want is to let your family members view your albums, you can publish (in album detail screen's menu) it to them. If you publish the album as Joint album, others can make addition to the album too.

**Comment by steviehs** [comment]
So, this icon is just an additional visualization as I see the shared folders also in my "main screen" in lespas?
Still talking about 400 shared folders - and maybe this from _every_ member of a family - would excess the action of manually setting and localizing folders. So the option to check multiple "lespas*" folders on the Nextcloud Server would be really great.

BTW, I should perhaps open up an other issue for that: selecting multiple folders and setting them as local is greyed out after I selected around more than 10 folders at a time?

**Comment by scubajeff** [comment]
Still trying to understand what exactly is the problem you are facing? 

Each member in your family group can publish his/her albums to the others. Each member can view the others' publications in the "Friend's Publication". Your own albums are in lespas/ folder, your received publications are in shared_with_me/ folder. And I believe Nextcloud manage to avoid duplication by using some mechanism similar to symbolic link. Isn't that already clean and tidy?

Or is it the effort to published each one of those albums a burden?

Or you would like to have one giant single list of all the albums of all the menbers?

About the missing menu item, make sure your selection doesn't mix local and remote albums.

**Comment by scubajeff** [comment]
By the way, I'm planning to add a advanced grouping function, e.g., user can choose to group photos in an album by time. So you might be able to have only a few albums by year, and within those albums, Lespas will group them by months for you automatically.

**Comment by steviehs** [comment]
is there a way to make the albums in the shared_with_me folder stored locally? I have not discovered that. But then it could be a way to go. e.g. I have a family tablet and want to have my personal shared albums on that, but - of course - stored locally as in the old days of lespas.

I will play around with the shared with me folder - fun fact is, that most of the users I am providing next cloud - even don't know about Nextclouds Web Interface...

The advanced grouping sounds nice, looking forward to it! (still hoping for the "store locally as default" feature soon :-)

**Comment by scubajeff** [comment]
Would a **local** "friend's publication" solve your problem?

**Comment by steviehs** [comment]
I will give the shared_with_me folders a try and if that works, I'd say yes :-) I mean, at the end, I could rename the "shared_with_me" folder to lespas and go without the publication icon. The drawback is that the NC users have to find other stuff shared with them also in this folder....

**Comment by scubajeff** [comment]
Share folder structure is controlled by NC. Lespas can't do anything about it.

**Comment by steviehs** [comment]
Ok, I'd say for my workflow the way to go is the following:
I set the "shared_with_me" folder name to "lespas".
I change my phonepics script to add a suffix to each YYYY_MM folder e.g. 202205_st
The automatic sharing is already part of the script.
Now every family members shared albums are in visible in the usual lespas album space.

An alternative would be to go with the "Friends Publication" and rename the shared_with_me folder to another name... this would end up in a different way to get access to the albums... dunno yet, what would be nicer for the viewer, I'd say the "lespas" (1st) approach.

Now the only thing I am missing to rework my folders for this is the "set local by default" option for the normal lespas folder... I could also write this down as a cookbook (at least for people who want to go this way) for your documentation.

**Comment by scubajeff** [comment]
In that sense, you don't need sharing/publishing at all. All members can logon to the same account in LesPas. You can make separate app key for each one so that you don't have to share your password to them.

**Comment by steviehs** [comment]
No, of course I want all members also to have their private albums, which they also put in their lespas folders but do not share them.
